### ISSUE TITLE
iOS: playback stops after 5-10s

### ISSUE THREAD

**Issue by andrewmwhite** [advplyr/audiobookshelf-app#933]
### Steps to reproduce

1. Open app
2. Select audiobook
3. Touch play button to start playback

### Expected behaviour
- playback continues indefinitely

### Actual behaviour
- playback stops after 5-10s
- starting a second time seems to proceed as normal
- not just paused -- iOS seems to lose the context as well, i.e., the book drops out of "now playing" so can't be resumed with iOS controls or Bluetooth device
- definitely happens with local books, I think also with streamed
- consistent; happens on every app start AFAICT

### Environment data

Audiobookshelf Version:

- [ ] Android App?
- [x] iOS App?

#### Android Issue
Android version:

Device model:

Stock or customized system:

#### iOS Issue
iOS Version: 0.9.66-beta

iPhone model: 14 Pro Max

**Comment by benonymity** [comment]
This is related to a bunch of other issues, but is mostly because it takes the app a second to connect to the server, so when you open it and immediately start playing a book, it is still attempting a server connection, and when it succeeds a few seconds later, the app doesn't know how to gracefully transition the existing offline session to being online. The temporary fix is to give the app a few seconds to catch up after you open it before pressing play, but when some other iOS playback bugs get fixed this'll also be resolved

**Comment by advplyr** [comment]
@benonymity I'm trying to reproduce this on v0.9.66 and server v2.5.0 but unable. I added a timeout in the server connection logic to ensure I could start listening before it finishes connecting. 

When first launching the app you only see the locally downloaded media. In testing this I'm playing one of the locally downloaded books before the server connection finishes. Once the connection finishes nothing happens with the playback, it carries on like normal.

Is there a step by step I can do to reproduce this issue?

**Comment by benonymity** [comment]
@advplyr this is what the issue looks like for me:
https://github.com/advplyr/audiobookshelf-app/assets/62854267/b6a541ac-4a85-461a-bc09-6a070bedee89

**Comment by advplyr** [comment]
I'm not sure why I can't reproduce that.
@andrewmwhite Can you confirm that this is what you experiencing?

**Comment by benonymity** [comment]
It could be that I'm 800 miles from my server, so there is a second of delay before it connects. Maybe you can always connect fast enough to avoid the issue?

**Comment by benonymity** [comment]
It could be that I'm 800 miles from my server, so there is a second of delay before it connects. Maybe you can always connect fast enough to avoid the issue?

**Comment by advplyr** [comment]
I added a delay in the code so that the connection takes 10 seconds. Unless this is only happening on production releases.

**Comment by andrewmwhite** [comment]
Mines a little different, more like 10s. The stops-on-server-connect-or-timeout hypothesis makes some sense to me though. I'll try to pay more attention to whether I let the app connect/timeout first or not. My server isn't internet accessible so it's often not connectable.

I did get a screen recording but it's not uploading... likely bad service where I am right now. Will give it another shot when I have reliable internet again.

**Comment by andrewmwhite** [comment]
Mines a little different, more like 10s. The stops-on-server-connect-or-timeout hypothesis makes some sense to me though. I'll try to pay more attention to whether I let the app connect/timeout first or not. My server isn't internet accessible so it's often not connectable.

I did get a screen recording but it's not uploading... likely bad service where I am right now. Will give it another shot when I have reliable internet again.

**Comment by andrewmwhite** [comment]
https://github.com/advplyr/audiobookshelf-app/assets/1322614/b5c9ac34-51bb-49f5-a7f4-9837b1aff01f

**Comment by andrewmwhite** [comment]
Here I don't start playback until after the connection is established and the library details load, but playback stops after a few seconds.

https://github.com/advplyr/audiobookshelf-app/assets/1322614/eb7199c3-935c-450b-8b63-9cab280de1ea

**Comment by advplyr** [comment]
Thanks for sharing the videos. I haven't been able to reproduce the issue so far and I think you are the first to bring up this specific issue. 

There is another issue users are having that has to do with pausing the app while it is in the background (phone locked) and the media player closing.

It would be best to try starting with a fresh install of the app where you have no potential database issues. However, since the beta is full right now reinstalling might lose your spot. If there is another way on iOS to clear all app data that would be great.

**Comment by benonymity** [comment]
I think you can actually just delete the app from your phone and still have it "checked out" in TestFlight, and just redownload it without getting re-added to the beta. I did that a few weeks ago without issues.

**Comment by andrewmwhite** [comment]
I was able to reinstall by deleting the app from the Home Screen then
installing again from TestFlight.

Will update if I notice the issue reoccurring.

Sent from my mobile device; please excuse the brevity, punctuation, and
spelling.

**Comment by gh0sti** [comment]
> I think you can actually just delete the app from your phone and still have it "checked out" in TestFlight, and just redownload it without getting re-added to the beta. I did that a few weeks ago without issues.

Once you are enrolled in a testflight beta you can delete and reinstall. I am enrolled in serveral apps where I can test from testflight and update to production from app store.

**Comment by andrewmwhite** [comment]
Seems to be resolved with the reinstall.
